### ISSUE TITLE
"Docker Quick Start Terminal" shortcut, an error occurs.

### ISSUE THREAD

**Issue by sabzil** [docker/toolbox#335]
Description of problem:

When you install the "DockerToolbox" for Windows and run "Docker Quickstart Terminal" shortcut, an error occurs.
![err](https://cloud.githubusercontent.com/assets/2362408/11583051/05a066ba-9a99-11e5-981b-dfc25ec155c3.png)

Set the "Docker Quickstart Terminal" shortcut icon "docker toolbox" for Windows are as follows.
![property](https://cloud.githubusercontent.com/assets/2362408/11583064/1fa1914c-9a99-11e5-8e4c-f541015b7800.png)

"C:\Program Files\Git\bin\bash.exe" --login -i "C:\Program Files\Docker Toolbox\start.sh"
This content was created by "Toolbox.iss", as follows:
Name: "{commondesktop}\Docker Quickstart Terminal"; WorkingDir: "{app}"; Filename: "{pf64}\Git\bin\bash.exe"; Parameters: "--login -i ""{app}\start.sh"""; IconFilename: "{app}/docker-quickstart-terminal.ico"; Tasks: desktopicon; Components: "Docker"

It is used in the "Filename" is to specify the bash.exe that exist in the path of git is installed.

This "Filename" value of the script is specified as "{pf64}\Git\bin\bash.exe".
Git the installation may be installed in a different path, like "%localappdata%\Programs\Git".

So the registry "HKLM{64|32}\Software\GitForWindows" or "HKCU{64|32}\SOFTWARE\GitForWindows" of "InstallPath" seems to confirm the more it would be fun to set the execution path of bash.exe.

`docker version`:
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   a34a1d5
 Built:        Fri Nov 20 17:56:04 UTC 2015
 OS/Arch:      windows/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   a34a1d5
 Built:        Fri Nov 20 17:56:04 UTC 2015
 OS/Arch:      linux/amd64

`docker info`:
Containers: 0
Images: 0
Server Version: 1.9.1
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 0
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.1.13-boot2docker
Operating System: Boot2Docker 1.9.1 (TCL 6.4.1); master : cef800b - Fri Nov 20 19:33:59 UTC 2015
CPUs: 1
Total Memory: 996.2 MiB
Name: default
ID: FU6B:RMRB:4BB6:OW4X:YOI6:YB6W:6X2L:X3MH:A7UO:UTPX:4O7S:COGD
Debug mode (server): true
 File Descriptors: 11
 Goroutines: 18
 System Time: 2015-12-04T05:43:35.843404146Z
 EventsListeners: 0
 Init SHA1:
 Init Path: /usr/local/bin/docker
 Docker Root Dir: /mnt/sda1/var/lib/docker
Labels:
 provider=virtualbox

`uname -a`:
MINGW64_NT-10.0 DESKTOP-6DD6KQ7 2.3.0(0.290/5/3) 2015-09-29 10:48 x86_64 Msys

Environment details (AWS, VirtualBox, physical, etc.):
Windows 10, DockerToolbox-1.9.1a.exe

How reproducible:

Steps to Reproduce:
1. Install the Docker Toolbox-1.9.1a.exe.
2. Double click on the "Docker Quickstart Terminal" shortcut on the desktop.

Actual Results:
an error window will appear that it can not find the bash.exe.

Expected Results:
Docker appears terminals.

Additional info:

**Comment by lukemac1** [comment]
I am experiencing what may be the same issue in Windows 10. I get the error "Missing Shortcut: Windows is searching for bash.exe." 
![image](https://cloud.githubusercontent.com/assets/10284961/11772199/e89b05ec-a1e7-11e5-90bb-c512c9f07905.png)

Windows searches around and finds a bash.exe in another program (GRASS), but if I choose 'browse', I can find the bash.exe in the 'bin' sub-directory of Git. Manually choosing bash.exe there allowed it to run docker on my first trial, but I'm having a new problem in docker now, which is probably unrelated, but just in case I'll include here:

docker run hello-world
 "an error occurred trying to connect: Post http://...: dial tcp 127....: ConnectEx tcp: No connection could be made because the target machine actively refused it."

**Comment by icnocop** [comment]
You'll have to edit the shortcut to resolve the issue with bash.exe.

On Windows 7:
1. Right-click on the shortcut
2. Select Properties
3. Edit the Target field to point to the location of bash.exe
ex. `C:\Users\username\AppData\Local\Programs\Git\bin\bash.exe --login -i "C:\Program Files\Docker Toolbox\start.sh"`
4. Click OK

**Comment by icnocop** [comment]
@lukemac1, In regards to your second issue, is it trying to connect to localhost? If so, see https://github.com/docker/docker/issues/17871.

**Comment by lukemac1** [comment]
Thank you @icnocop. That did it. After a restart, the second issue i raised about the ConnectEx tcp connection problem has resolved on its own for some reason.

**Comment by jaehoonkim** [comment]
Thank you @icnocop .

**Comment by funkygibbing** [comment]
I think closing this is premature, as it's a bug that completely breaks the docker installation process.

**Comment by ramiabughazaleh** [comment]
See issue https://github.com/docker/toolbox/issues/343 and pull request https://github.com/docker/toolbox/pull/324.

**Comment by jaehoonkim** [comment]
Sorry. Place open it again, should I issue # 343 and # 324 as it closes when it closes?

**Comment by burzum** [comment]
This is still an issue, I just ran into the same problem.

**Comment by sharsh** [comment]
It worked for me! 
First Installing git by selecting 'Run Git and included unix tools from windows command prompt' and then installing docker tool box without selecting GIT from the list

**Comment by jaehoonkim** [comment]
@sharsh Thank you. 
Did not fundamentally resolved, I'm sorry closing the issue.
I will reopen this issue.

**Comment by rojagit** [comment]
This is just an instance of unselecting Git and having docker search for git on its own.
They grandly say do not install VB but I didnt want to take any chances this time so
after facing numerous installation issues on Windows I uninstalled Git & VB and am retrying. 
Last time everything went through but in the end it said docker.exe NF. 
Ok success this time install looks good.

**Comment by tfjwr** [comment]
Same bug after 6 months. After some 3 hours of lost time, didn't have to delete VirtuaBox in my case, but deleting Git for windows before installing docker toolbox for windows with Git, ( also VirtualBox selected in the option, nonetheless ) seems to have worked.

**Comment by alturium** [comment]
Updated the shortcut to use the GIT bash.exe, and noticed that the the shortcut referenced C; vice D: (where I had installed DockerTools).

Fix:
"D:\Program Files\Git\bin\bash.exe" --login -i "D:\Program Files\Docker Toolbox\start.sh"

so it appears to be a install issue.

**Comment by eromoe** [comment]
"D:\Program Files\Git\bin\bash.exe" --login -i "D:\Program Files\Docker Toolbox\start.sh"

1. can not select text copy and paste,  can not change to quickedit mode 
2. no context menu like   `open git-bash here`

It is hurt.

**Comment by eromoe** [comment]
Is anyway that just enter `docker-start`   to go into docker terminal mode?

**Comment by Nilegfx** [comment]
I had the same issue in my machine, I changed **Program Files** to **Program Files (x86)** and everything worked just fine.

**Comment by rmorse** [comment]
Path was incorrect for me also, just searched for `bash.exe` on my machine and pointed the shortcut to that

**Comment by justnx** [comment]
Check if you had installed GIT for Windows as Standalone package before. If so, uninstall GIT and start Docker Installation process again, then it should work fine.

**Comment by sabzil** [comment]
Thank you. It is good to delete git and install the docker first.

**Comment by rmorrise** [comment]
If this is a prerequisite for a docker toolbox install, the docker toolbox installer should do it. Don't require users to install the tool, google the issue, uninstall the tool, uninstall a different tool, and reinstall the tool.

**Comment by gibranamparan** [comment]
Because i have previously installed git on C:\Program Files (x86)\Git\ I had to point the shortcut to C:\Program Files (x86)\Git\bin\bash.exe and it worked. So the target in shotcut properties worked as "C:\Program Files\Git\bin\bash.exe" --login -i "D:\Program Files\Docker Toolbox\start.sh"

**Comment by Download** [comment]
I saw the installer remove Git then install Git for Windows, but alas I still experienced this issue when first trying to open Docker Quickstart Terminal.

**Comment by gety9** [comment]
The issue was open on Dec 4, 2015, and not it's 2018 and it's still there.

**Comment by chisumo2016** [comment]
Error creating machine: Error in driver during machine creation: This computer d
oesn't have VT-X/AMD-v enabled. Enabling it in the BIOS is mandatory
Looks like something went wrong in step ´Checking if machine default exists´...
Press any key to continue...    .My Laptop is running on window 7

**Comment by GowthamS07** [comment]
Oh it is 2020 and the issue is still here !!!

**Comment by ragvri** [comment]
@icnocop you da man!

**Comment by Kaweees** [comment]
It is still an issue for me

**Comment by Download** [comment]
There are half a dozen people here reporting this is still an issue. You might want to re-open it.

Since this is an issue in the installer, people that are trying out Docker are turned away right at their first try.
I know that I have not used Docker since I ran into this installer issue. If I now see instructions to use something with Docker, I think: 'won't work for me since I can't get Docker to install on my machine' and then pick some other option. After initially being very enthusiastic about Docker, I basically have never considered using it again since I got this issue and saw it not being addressed.
